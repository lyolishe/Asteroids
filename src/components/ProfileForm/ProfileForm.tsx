import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

import {
  Avatar, Form, Input, Button, Row, Col, Card,
} from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { useForm } from 'antd/es/form/Form';

import AuthApi from '../../api/AuthApi';
import UserApi from '../../api/UserApi';

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 16 },
};
const tailLayout = {
  wrapperCol: { offset: 8, span: 16 },
};

interface FormValues {
  first_name?: string,
  second_name?: string,
  display_name?: string,
  login?: string,
  phone?: string,
  email?: string,
  oldPassword?: string,
  newPassword?: string
}

const ProfileForm = (): JSX.Element => {
  const [form] = useForm();

  const [avatar, setAvatar] = useState('');

  useEffect(() => {
    AuthApi.getUser().then((user) => {
      setAvatar(`https://ya-praktikum.tech${user.avatar}`);

      form.setFieldsValue({
        first_name: user.first_name,
        second_name: user.second_name,
        display_name: user.display_name,
        login: user.login,
        phone: user.phone,
        email: user.email,
      });
    });
  });

  const onFinish = (values: FormValues) => {
    UserApi.editProfile(values);

    if (values.oldPassword && values.newPassword) {
      UserApi.changePassword({
        oldPassword: values.oldPassword,
        newPassword: values.newPassword,
      });
    }
  };

  const onAvatarUpload = (e: React.ChangeEvent<HTMLInputElement>) : void => {
    if (e.target.files) {
      UserApi.uploadAvatar(e.target.files[0]);
    }
  };

  return (
    <Form
      {...layout}
      name="basic"
      onFinish={onFinish}
      layout="vertical"
      hideRequiredMark
      form={form}
    >
      <Row>
        <Col span={12} offset={6}>
          <Card title="Профиль пользователя">
            {avatar ? <Avatar shape="square" size={64} src={avatar} /> : <Avatar shape="square" size={64} icon={<UserOutlined />} />}
            <Form.Item
              label="Аватар"
              name="avatar"
            >
              <Input type="file" onChange={onAvatarUpload} />
            </Form.Item>
            <Form.Item
              label="Отображаемое имя"
              name="display_name"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>
            <Form.Item
              label="Имя"
              name="first_name"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Фамилия"
              name="second_name"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Логин"
              name="login"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Почта"
              name="email"
              rules={[{ required: true, type: 'email', message: 'Неверный email' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Телефон"
              name="phone"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input maxLength={12} />
            </Form.Item>

            <Form.Item
              label="Старый пароль"
              name="oldPassword"
            >
              <Input.Password />
            </Form.Item>

            <Form.Item
              label="Новый пароль"
              name="newPassword"
            >
              <Input.Password />
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Link to="/">
                <Button htmlType="button">
                  На главную
                </Button>
              </Link>
              <Button type="primary" htmlType="submit">
                Сохранить
              </Button>
            </Form.Item>
          </Card>
        </Col>
      </Row>
    </Form>
  );
};

export default ProfileForm;
