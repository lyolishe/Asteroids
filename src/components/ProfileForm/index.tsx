import React, { useState, useEffect } from "react";
import {Redirect} from 'react-router-dom';

import {Avatar, Form, Input, Button, Row, Col, Card, Select} from 'antd';
import { UserOutlined } from '@ant-design/icons';
import {useForm} from "antd/es/form/Form";

import AuthApi from "../../api/AuthApi";

const {Option} = Select;

const layout = {
    labelCol: {span: 8},
    wrapperCol: {span: 16},
};
const tailLayout = {
    wrapperCol: {offset: 8, span: 16},
};

const ProfileForm = () => {
    const [form] = useForm();

    useEffect(() => {
      AuthApi.getUser().then(r => {
        form.setFieldsValue({
          avatar: r.avatar,
          first_name: r.first_name,
          second_name: r.second_name,
          display_name: r.display_name,
          login: r.login,
          phone: r.phone,
          email: r.email,
        });
      }).catch(error => {
        console.log(error);
      });
    }, []);

    const onFinish = (values: any) => {
        console.log('Success:', values);
    };

    const onFinishFailed = (errorInfo: any) => {
        console.log('Failed:', errorInfo);
    };

    const toMain = () => {
        return <Redirect to="/" />
    }

    const prefixSelector = (
        <Form.Item name="prefix" noStyle>
            <Select style={{width: 70}}>
                <Option value="375">+375</Option>
                <Option value="7">+7</Option>
            </Select>
        </Form.Item>
    );

    return (
        <Form
            {...layout}
            name="basic"
            onFinish={onFinish}
            onFinishFailed={onFinishFailed}
            layout={'vertical'}
            hideRequiredMark
            form={form}
        >
            <Row>
                <Col span={12} offset={6}>
                    <Card title={'Профиль пользователя'}>
                    	<Avatar shape="square" size={64} icon={<UserOutlined />} />
                    	<Form.Item
                            label="Аватар"
                            name="avatar"
                        >                        	
                            <Input type="file"/>
                        </Form.Item>
                        <Form.Item
                            label="Имя"
                            name="first_name"
                            rules={[{required: true, message: 'Заполните поле!'}]}
                        >
                            <Input/>
                        </Form.Item>

                        <Form.Item
                            label="Фамилия"
                            name="second_name"
                            rules={[{required: true, message: 'Заполните поле!'}]}
                        >
                            <Input/>
                        </Form.Item>

                        <Form.Item
                            label="Логин"
                            name="login"
                            rules={[{required: true, message: 'Заполните поле!'}]}
                        >
                            <Input/>
                        </Form.Item>

                        <Form.Item
                            label="Почта"
                            name="email"
                            rules={[{required: true, type: 'email', message: 'Неверный email'}]}
                        >
                            <Input/>
                        </Form.Item>

                        <Form.Item
                            label="Телефон"
                            name="phone"
                            rules={[{required: true, message: 'Заполните поле!'}]}
                        >
                            <Input maxLength={9} addonBefore={prefixSelector}/>
                        </Form.Item>

                        <Form.Item
                            label="Пароль"
                            name="password"
                            rules={[{required: true, message: 'Введите пароль!'}]}
                        >
                            <Input.Password/>
                        </Form.Item>

                        <Form.Item {...tailLayout}>
                        	<Button htmlType="button" onClick={toMain}>
                                Назад
                            </Button>
                            <Button type="primary" htmlType="submit">
                                Сохранить
                            </Button>
                        </Form.Item>
                    </Card>
                </Col>
            </Row>
        </Form>
    );
};

export default ProfileForm;