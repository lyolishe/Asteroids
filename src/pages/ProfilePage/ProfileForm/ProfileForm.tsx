// Core
import React from 'react';

// Components
/*import { Avatar, Form, Input } from 'antd';
import { UserOutlined } from '@ant-design/icons';*/

// Api
import UserApi from '../../../api/UserApi';

// Types
export interface ProfileFormProps {
  avatar: string;
}

const ProfileForm: React.FC<ProfileFormProps> = ({
  avatar,
}) => {
  const onAvatarUpload = (e: React.ChangeEvent<HTMLInputElement>) : void => {
    if (e.target.files) {
      UserApi.uploadAvatar(e.target.files[0]);
    }
  };

  return (
    <>
      {avatar ? <Avatar shape="square" size={64} src={avatar} /> : <Avatar shape="square" size={64} icon={<UserOutlined />} />}
      <Form.Item
        label="Аватар"
        name="avatar"
      >
        <Input type="file" onChange={onAvatarUpload} />
      </Form.Item>
      <Form.Item
        label="Отображаемое имя"
        name="display_name"
        rules={[{ required: true, message: 'Заполните поле!' }]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        label="Имя"
        name="first_name"
        rules={[{ required: true, message: 'Заполните поле!' }]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        label="Фамилия"
        name="second_name"
        rules={[{ required: true, message: 'Заполните поле!' }]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        label="Логин"
        name="login"
        rules={[{ required: true, message: 'Заполните поле!' }]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        label="Почта"
        name="email"
        rules={[{ required: true, type: 'email', message: 'Неверный email' }]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        label="Телефон"
        name="phone"
        rules={[{ required: true, message: 'Заполните поле!' }]}
      >
        <Input maxLength={12} />
      </Form.Item>

      <Form.Item
        label="Старый пароль"
        name="oldPassword"
      >
        <Input.Password />
      </Form.Item>

      <Form.Item
        label="Новый пароль"
        name="newPassword"
      >
        <Input.Password />
      </Form.Item>
    </>
  );
};

export default ProfileForm;
