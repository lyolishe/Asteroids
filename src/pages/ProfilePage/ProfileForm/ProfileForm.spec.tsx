import React from 'react';

import { createRender } from '@material-ui/core/test-utils';
import { render, RenderResult } from '@testing-library/react';
import ProfileForm from './ProfileForm';

let documentBody: RenderResult;

describe('<ProfileForm />', () => {
  beforeEach(() => {
    documentBody = render(<ProfileForm />);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
