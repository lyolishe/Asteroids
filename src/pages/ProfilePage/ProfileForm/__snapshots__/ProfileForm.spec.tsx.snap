// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileForm /> matches snapshot 1`] = `
<body>
  <div>
    <form
      novalidate=""
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-7"
      >
        <input
          accept="image/*"
          class="makeStyles-hidden-6"
          id="avatar-input"
          name="avatar"
          type="file"
        />
        <label
          class="makeStyles-avatarButton-3"
          for="avatar-input"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Загрузить аватар
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </label>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Отображаемое имя
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="display_name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Отображаемое имя
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Имя
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="first_name"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Имя
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Фамилия
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="second_name"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Фамилия
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Логин
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="login"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Логин
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Почта
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="email"
            required=""
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Почта
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Телефон
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="phone"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Телефон
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Старый пароль
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="oldPassword"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Старый пароль
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Новый пароль
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="newPassword"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-10"
            >
              <span>
                Новый пароль
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Сохранить
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="/"
          >
            Играть
          </a>
        </div>
      </div>
    </form>
  </div>
</body>
`;
